res/controllers/Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -0,0 +1,772 @@
+        midi.sendShortMsg(0x94 + i, 0x37, 0x01); // double
+        midi.sendShortMsg(0x94 + i, 0x41, 0x01); // loop
+
+        // pads
+        for (var j = 0; j < 8; j++) {
+            midi.sendShortMsg(0x94 + i, 0x14 + j, 0x01);
+            midi.sendShortMsg(0x94 + i, 0x1C + j, 0x01); // shifted
+        }
+    }
+
+    midi.sendShortMsg(0x88, 0x09, 0x01); // tap led
+
+    // check if quantize is enabled
+    var quantizeEnabled = engine.getValue("[Channel1]", "quantize");
+
+    // effect leds
+    midi.sendShortMsg(0x88, 0x00, 0x01); // hpf
+    midi.sendShortMsg(0x88, 0x01, 0x01); // lpf
+    midi.sendShortMsg(0x88, 0x02, 0x01); // flanger
+    midi.sendShortMsg(0x89, 0x03, 0x01); // echo
+    midi.sendShortMsg(0x89, 0x04, quantizeEnabled ? 0x7F : 0x01); // reverb
+    midi.sendShortMsg(0x89, 0x05, 0x01); // phaser
+
+    // vumeters leds (off)
+    midi.sendShortMsg(0xB0, 0x1F, 0x00);
+    midi.sendShortMsg(0xB1, 0x1F, 0x00);
+
+    engine.makeConnection("[Channel1]", "VuMeter", MixtrackProFX.vuCallback);
+    engine.makeConnection("[Channel2]", "VuMeter", MixtrackProFX.vuCallback);
+};
+
+MixtrackProFX.shutdown = function() {
+    var shutdownSysex = [0xF0, 0x00, 0x20, 0x7F, 0x02, 0xF7];
+    midi.sendSysexMsg(shutdownSysex, shutdownSysex.length);
+};
+
+MixtrackProFX.EffectUnit = function(unitNumber) {
+    this.unitNumber = unitNumber;
+    this.group = "[EffectRack1_EffectUnit" + unitNumber + "]";
+
+    this.enableButton = new components.Button({
+        input: function(channel, control, value, status, group) {
+            if (value === 2) {
+                value = 1;
+            }
+            engine.setValue(group, "enabled", value);
+        }
+    });
+
+    this.dryWetKnob = new components.Pot({
+        group: this.group,
+        inKey: "mix"
+    });
+
+    this.tap = new components.Button({
+        group: "[Channel" + this.unitNumber + "]",
+        key: "bpm_tap",
+        midi: [0x88, 0x09],
+        off: 0x01
+    });
+
+    this.effectParam = new components.Encoder({
+        group: "[EffectRack1_EffectUnit" + unitNumber + "_Effect1]",
+        inKey: "parameter1",
+        shift: function() {
+            this.inKey = "parameter2";
+        },
+        unshift: function() {
+            this.inKey = "parameter1";
+        },
+        input: function(channel, control, value) {
+            if (value === 0x01) {
+                this.inSetParameter(this.inGetParameter() + 0.05);
+            } else if (value === 0x7F) {
+                this.inSetParameter(this.inGetParameter() - 0.05);
+            }
+        }
+    });
+};
+
+MixtrackProFX.EffectUnit.prototype = new components.ComponentContainer();
+
+MixtrackProFX.Deck = function(number) {
+    var deck = this;
+    var channel = number - 1;
+    var blinkTimer = 0;
+    var blinkLedState = true;
+
+    components.Deck.call(this, number);
+
+    this.playButton = new components.PlayButton({
+        midi: [0x90 + channel, 0x00],
+        off: 0x01
```

Comment:
Nice, thank you for the tip.

Sending 0x01 is not because the controller interprets 0x01 as led off (it interprets 0x00 as led off). It's to make the lights dim when in inactive state, instead of turning them off. This way you can see where the buttons are in darkness.

Reply:
That should be documented in a comment.